libraries/c_sdk/standard/ble/include/iot_ble_wifi_provisioning.h

```diff
@@ -32,69 +32,103 @@
+ * @brief Defines the list wifi networks request message structure sent from the provisioining app to the device.
+ * The request is used to list already provisioned networks as well as to scan nearby access points.
  */
-typedef struct IotBleListNetworkRequest
+typedef struct
 {
-    int16_t maxNetworks; /**< Max Networks to scan in one request */
-    int16_t timeoutMs;   /**< Timeout in MS for scanning */
-} IotBleListNetworkRequest_t;
+    int16_t scanSize;      /**< Max WiFi access points to scan in one request */
+    int16_t scanTimeoutMS; /**< Timeout in milliseconds for scanning */
+} IotBleWifiProvListNetworksRequest_t;
 
 /**
  * @ingroup ble_datatypes_structs
- * @brief Sent by the GATT client to provision a new WiFi network.
+ * @brief Defines add wifi network request message structure sent from the provisioining app to the device.
+ * The request is used to provision a new access point with the credentials or to connect to an already provisioned
+ * access point. Inorder to connect to an already provisioned access point, a valid accessPointStoredIndex needs
+ * to be provided. Setting flag connectToAccessPoint to true results in the  device first connecting to access point successfully before
+ * provisioning it in the flash.
  */
-typedef struct IotBleAddNetworkRequest
+typedef struct
 {
-    WIFINetworkProfile_t network; /**< The configuration for the new WIFI network. */
-    int16_t savedIdx;             /**< Index if its an already saved WIFI network in the flash. */
-    bool connect;                 /**< A flag to indicate whether to connect or not. */
-} IotBleAddNetworkRequest_t;
+    union
+    {
+        WIFINetworkProfile_t network; /**< The configuration for the new WIFI access point. */
+        int16_t index;                /**< Prirority index of an already provisioned WiFi access point. */
+    } info;
+    bool isProvisioned : 1;           /**< true if the newtwork is already provisioned, a valid pirority index should be provided. */
+    bool shouldConnect : 1;           /**< true if the access point needs to be connected before provisioning. */
+} IotBleWifiProvAddNetworkRequest_t;
 
 /**
  * @ingroup ble_datatypes_structs
- * @brief Sent by the GATT client to change the saved WiFi networks priority order.
+ * @brief Defines edit wifi network request message structure sent from provisioning app to the device. The request
+ * is used to change the priority index for a provisioned access point. Priority index ranges from 0 (Max priority) to
+ * wificonfigMAX_NETWORK_PROFILES - 1 (Minimum priority).
  */
-typedef struct IotBleEditNetworkRequest
+typedef struct
 {
-    int16_t curIdx; /**< Current priority of the saved WiFi network. */
-    int16_t newIdx; /**< New priority of the saved WiFi network. */
-} IotBleEditNetworkRequest_t;
+    int16_t currentPriorityIndex; /**< Current Prioruty index of the provisioned WiFi network.*/
```

Comment:
spelling: _Priority_ – is spell check enabled on this repo?

Reply:
Fixed. Currently spell check is not enabled on this repo.